client/views/view_units.cpp

```diff
@@ -217,44 +329,37 @@ void units_view::update_view()
 void units_view::selection_changed(const QItemSelection &sl,
                                    const QItemSelection &ds)
 {
-  // QTableWidgetItem *itm;
-  // int i;
+  QTableWidgetItem *itm;
   QVariant qvar;
-  // struct universal selected;
+  struct universal selected;
+  QString upg;
   // const struct unit_view_entry *pentry;
+
   ui.upg_but->setDisabled(true);
-  // These two are always available
-  ui.find_but->setDisabled(false);
-  ui.disband_but->setDisabled(false);
+  ui.disband_but->setDisabled(true);
+  ui.find_but->setDisabled(true);
 
   if (sl.isEmpty()) {
     return;
   }
 
   // TODO: Add code to enable the upgrade units button
-  /*curr_row = sl.indexes().at(0).row();
+  curr_row = sl.indexes().at(0).row();
   if (curr_row >= 0 && curr_row <= max_row) {
     itm = ui.units_widget->item(curr_row, 0);
     qvar = itm->data(Qt::UserRole);
     uid = qvar.toInt();
     selected = cid_decode(uid);
     counter = ui.units_widget->item(curr_row, 3)->text().toInt();
-      if (can_sell_building(pimprove)) {
-        ui.bsell->setEnabled(true);
-      }
-      itm = ui.eco_widget->item(curr_row, 2);
-      i = itm->text().toInt();
-      if (i > 0) {
-        ui.bredun->setEnabled(true);
-      }
-      break;
-    case VUT_UTYPE:
-      counter = ui.eco_widget->item(curr_row, 3)->text().toInt();
-      ui.bdisband->setEnabled(true);
-      break;
-    default:
-      qCritical("Not supported type: %d.", selected.kind);
-    }*/
+    if (counter > 0) {
+      ui.disband_but->setDisabled(false);
```

Comment:
Why can't we disband multiple unit types at once? If you want the user to only ever act on a single unit type, you can change the selection mode of the table to `SingleSelection`.

Reply:
`SingleSelection` is on by default. This is checking to see if there any of the unit active. I didn't code to do more than one mass disband at once. This is probably a new feature issue post merge.